src/SignalR/clients/csharp/Http.Connections.Client/src/HttpConnection.cs

```diff
@@ -529,42 +527,39 @@ private HttpClient CreateHttpClient()
             var httpClientHandler = new HttpClientHandler();
             HttpMessageHandler httpMessageHandler = httpClientHandler;
 
+            var isBrowser = RuntimeInformation.IsOSPlatform(OSPlatform.Create("browser"));
+
             if (_httpConnectionOptions != null)
             {
-                if (_httpConnectionOptions.Proxy != null)
+                if (!isBrowser)
                 {
-                    httpClientHandler.Proxy = _httpConnectionOptions.Proxy;
-                }
+                    // Configure options that do not work in the browser inside this if-block
+                    if (_httpConnectionOptions.Proxy != null)
+                    {
+                        httpClientHandler.Proxy = _httpConnectionOptions.Proxy;
+                    }
 
-                try
-                {
-                    // On supported platforms, we need to pass the cookie container to the http client
-                    // so that we can capture any cookies from the negotiate response and give them to WebSockets.
                     httpClientHandler.CookieContainer = _httpConnectionOptions.Cookies;
-                }
-                // Some variants of Mono do not support client certs or cookies and will throw NotImplementedException or NotSupportedException
-                // Also WASM doesn't support some settings in the browser
-                catch (Exception ex) when (ex is NotSupportedException || ex is NotImplementedException)
```

Comment:
How do you feel about removing this try-catch? Supporting Mono isn't really a thing and we do not use this code path in browser.

Reply:
This fails on WatchOS